base/src/com/google/idea/blaze/base/buildview/BazelProcess.kt

```diff
@@ -0,0 +1,42 @@
+package com.google.idea.blaze.base.buildview
+
+import com.google.common.util.concurrent.ListenableFuture
+import com.google.idea.blaze.exception.BuildException
+import com.intellij.execution.process.ProcessHandler
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.guava.asListenableFuture
+
+/**
+ * Represents a running bazel process with an eventual result. The actual result
+ * depends on the context (i.e. build returns BlazeBuildOutputs and test returns
+ * BlazeTestOutputs).
+ */
+data class BazelProcess<T>(val hdl: ProcessHandler, private val result: Deferred<T>) {
+
+  suspend fun getResult(): T {
+    return result.await()
+  }
+
+  fun getResultFuture(): ListenableFuture<T> {
```

Comment:
this can be private

Reply:
I wanted to leave this exposed for future use from java. Because `getResult` cannot be used from java.